src/commands/RPG/RPG.ts

```diff
@@ -0,0 +1,592 @@
+        if (defender.hp <= 0) {
+          break
+        }
+      }
+      rounds += 1
+    }
+
+    let victor, loser: Character
+    // Append the summary text to the log
+    if (challenger.hp <= 0) {
+      victor = accepter
+      loser = challenger
+    } else if (accepter.hp <= 0) {
+      victor = challenger
+      loser = accepter
+    } else {
+      // Must be a draw. Leave victor and loser undefined.
+      const summary = `After ${RPG.MAX_ROUNDS} rounds they decide to call it a draw.`
+      return { intro: intro, log: log, summary: summary, challenger: challenger, accepter: accepter }
+    }
+
+    log += '\n\n'
+    const summary: string = getRandomElement(victoryTexts)
+      .replace(/VICTOR/g, `${victor.user}`)
+      .replace(/LOSER/g, `${loser.user}`)
+    log += summary
+
+    const result = {
+      intro: intro,
+      log: log,
+      winner: victor,
+      loser: loser,
+      summary: summary,
+      challenger: challenger,
+      accepter: accepter,
+    }
+
+    return result
+  }
+
+  @Slash('rpg_character', { description: 'Print your RPG character sheet' })
+  async rpgCharacter(interaction: CommandInteraction) {
+    const callerMember = getCallerFromCommand(interaction)
+    const callingUser = callerMember?.user
+
+    if (callingUser) {
+      const userDBRecord = await this.getUserFromDB(callerMember.user.id)
+      const eloBandIcon = this.getBandForEloRank(userDBRecord.eloRank)
+      const character = new Character(callingUser, callerMember.nickname ?? undefined)
+      interaction.reply({ embeds: [character.toEmbed(eloBandIcon.icon)] })
+    } else {
+      interaction.reply('Username undefined')
+    }
+  }
+
+  @Slash('rpg_stats', { description: 'Display your RPG fight statistics' })
+  private async rpgStats(interaction: CommandInteraction) {
+    await interaction.deferReply()
+
+    const callerMember = interaction.member
+    if (callerMember && callerMember instanceof GuildMember) {
+      const callerDBRecord = await this.getUserFromDB(callerMember.user.id)
+      const eloBand = this.getBandForEloRank(callerDBRecord.eloRank)
+      const statsEmbed = new MessageEmbed()
+        .setColor('#009933') // Could dig out the user's colour?
+        .setAuthor({
+          iconURL: callerMember.user.avatarURL() ?? '',
+          name: `${callerMember.nickname ?? callerMember.user.username}'s prowess in the arena: ${
+            callerDBRecord.wins
+          }W ${callerDBRecord.losses}L ${callerDBRecord.draws}D`,
+        })
+        .setDescription(`**Ladder Points:** ${callerDBRecord.eloRank} - ${eloBand.icon} *${eloBand.name} League*`)
+      await interaction.followUp({ embeds: [statsEmbed] })
+    } else {
+      await interaction.followUp(`Hmm, ${interaction.user}... It seems you are yet to test your steel.`)
+    }
+  }
+
+  @Slash('rpg_challenge')
+  private async rpg_challenge(interaction: CommandInteraction) {
+    await interaction.deferReply()
```

Comment:
Due to discord api limitations not specifying ephemeral here will make the first followup non ephemeral. We can avoid this by removing this call and making the followups on line 257, 270, 282 and 324 replies. 

We get 3s before Discord errors due to no reply being sent which should be plenty of time for all of this to take place. This will also allow the error messages before the actual challenge to be ephemeral and any followup after that ephemerality will work as expected.

Reply:
Seems like a similar fix can also be made in `/duel` as well? Maybe I'm misunderstanding though

Reply:
Yeah this is a change we should probably make on duel. It will make the ephemeral errors consistent.

Reply:
Is it specifically only those 4 `followUp` that should be changed to `reply`? What about 332? There are others in the `on collect` event. Is this a different case?

Reply:
It just boils down to when messages are shown. The slash command api requires the output of a slash command to have a reply or followup. So you need to either do `deferReply` and `followup/editReply`s OR `reply` and `followup/editReply`. Discord returns an error when you attempt to defer and then reply without first rollowing up or editing the deferral. 

The issue with the current logic is an issue with other commands we have where if the initial response is an error it ignores the ephemeral flag since the initial deferral isn't ephemeral. In contrast you can make the initial deferral ephemeral but now all `followUp/editReply`s will also only be ephemeral. So the actual challenge with the button will be ephemeral.